scanpointgenerator/generators/powertermgenerator.py

```diff
@@ -0,0 +1,82 @@
+from scanpointgenerator.core import Generator
+from scanpointgenerator.compat import np
+
+
+@Generator.register_subclass("scanpointgenerator:generator/PowerTermGenerator:1.0")
+class PowerTermGenerator(Generator):
+    """
+    Generate a line of points according to the function
+        y = ((x-x0)/a)**n + y0
+    """
+
+    def __init__(self, axis, units, start, stop, exponent, divisor, offset):
+        """
+        y = ((x-x0)/a)**n + y0
+        Args:
+            axis: The scannable axis e.g. "dcm_energy"
+            units: The scannable units e.g. "keV"
+            start: The first position to be generated. Will determine x0 in the above equation
+            stop: Will determine scan size. The final generated position will not necessarily be this...
+            exponent: n in above equation
+            divisor: a in the above equation
+            offset: y0 in the above equation. Point of interest which will be most finely sampled
+                    e.g. 7.112 (for Fe K edge)
+        """
+
+        if start >= stop:
+            raise ValueError("Start value must be lower than stop value")
+
+        if exponent < 1 or exponent != int(exponent):
+            raise ValueError("Exponent must be a positive integer")
+
+        if exponent % 2 == 0 and start < offset:
+            raise ValueError("For even exponents, start value must be greater or equal to offset")
```

Comment:
Always implies you're coming from the "right" of your focus point.
If you wrote instead `f(x) = A*((x-x0)/a)**n+y0` where A could be 1 or -1 (set according to start-offset <=> 0?) then you could allow approaching from the left since. Need to make sure you pick the right root though.